File: mifosng-android/src/main/java/com/mifos/objects/ErrorSyncServerMessage.java
package com.mifos.objects;

import java.util.List;

/**
 * Created by Rajan Maurya on 23/07/16.
 */
public class ErrorSyncServerMessage {

    String developerMessage;

    int httpStatusCode;

    String defaultUserMessage;

    String userMessageGlobalisationCode;

    List<Error> errors;

    public String getDeveloperMessage() {
        return developerMessage;
    }

    public void setDeveloperMessage(String developerMessage) {
        this.developerMessage = developerMessage;
    }

    public int getHttpStatusCode() {
        return httpStatusCode;
    }

    public void setHttpStatusCode(int httpStatusCode) {
        this.httpStatusCode = httpStatusCode;
    }

    public String getDefaultUserMessage() {
        return defaultUserMessage;
    }

    public void setDefaultUserMessage(String defaultUserMessage) {
        this.defaultUserMessage = defaultUserMessage;
    }

    public String getUserMessageGlobalisationCode() {
        return userMessageGlobalisationCode;
    }

    public void setUserMessageGlobalisationCode(String userMessageGlobalisationCode) {
        this.userMessageGlobalisationCode = userMessageGlobalisationCode;
    }

    public List<Error> getErrors() {
        return errors;
    }

    public void setErrors(List<Error> errors) {
        this.errors = errors;
    }

    public class Error {

        String developerMessage;

        String defaultUserMessage;

        String userMessageGlobalisationCode;

        String parameterName;

        public String getDeveloperMessage() {
            return developerMessage;
        }

        public void setDeveloperMessage(String developerMessage) {
            this.developerMessage = developerMessage;
        }

        public String getDefaultUserMessage() {
            return defaultUserMessage;
        }

        public void setDefaultUserMessage(String defaultUserMessage) {
            this.defaultUserMessage = defaultUserMessage;
        }

        public String getUserMessageGlobalisationCode() {
            return userMessageGlobalisationCode;
        }

        public void setUserMessageGlobalisationCode(String userMessageGlobalisationCode) {
            this.userMessageGlobalisationCode = userMessageGlobalisationCode;
        }

        public String getParameterName() {
            return parameterName;
        }

        public void setParameterName(String parameterName) {
            this.parameterName = parameterName;
        }
    }
}

File: core/model/src/main/kotlin/com/mifos/core/modelobjects/ErrorSyncServerMessage.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.core.modelobjects

/**
 * Created by Rajan Maurya on 23/07/16.
 */
class ErrorSyncServerMessage {
    var developerMessage: String? = null
    var httpStatusCode = 0
    var defaultUserMessage: String? = null
    var userMessageGlobalisationCode: String? = null
    var errors: List<Error>? = null

    inner class Error {
        var developerMessage: String? = null
        var defaultUserMessage: String? = null
        var userMessageGlobalisationCode: String? = null
        var parameterName: String? = null
    }
}
